src/server/frontend_wayland/xdg_shell_unstable_v6.cpp

```diff
@@ -0,0 +1,450 @@
+{
+    MirResizeEdge mir_edges = mir_resize_edge_none;
+
+    switch (edges)
+    {
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_TOP:
+        mir_edges = mir_resize_edge_north;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_BOTTOM:
+        mir_edges = mir_resize_edge_south;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_LEFT:
+        mir_edges = mir_resize_edge_west;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_TOP_LEFT:
+        mir_edges = mir_resize_edge_northwest;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_BOTTOM_LEFT:
+        mir_edges = mir_resize_edge_southwest;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_RIGHT:
+        mir_edges = mir_resize_edge_east;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_TOP_RIGHT:
+        mir_edges = mir_resize_edge_northeast;
+        break;
+
+    case ZXDG_TOPLEVEL_V6_RESIZE_EDGE_BOTTOM_RIGHT:
+        mir_edges = mir_resize_edge_southeast;
+        break;
+
+    default:;
+    }
+
+    base->resize(seat, serial, mir_edges);
+}
+
+void mf::XdgToplevelUnstableV6::set_max_size(int32_t width, int32_t height)
+{
+    base->set_max_size(width, height);
+}
+
+void mf::XdgToplevelUnstableV6::set_min_size(int32_t width, int32_t height)
+{
+    base->set_min_size(width, height);
+}
+
+void mf::XdgToplevelUnstableV6::set_maximized()
+{
+    base->set_maximized();
+}
+
+void mf::XdgToplevelUnstableV6::unset_maximized()
+{
+    base->unset_maximized();
+}
+
+void mf::XdgToplevelUnstableV6::set_fullscreen(std::experimental::optional<struct wl_resource*> const& output)
+{
+    base->set_fullscreen(output);
+}
+
+void mf::XdgToplevelUnstableV6::unset_fullscreen()
+{
+    base->unset_fullscreen();
+}
+
+void mf::XdgToplevelUnstableV6::set_minimized()
+{
+    base->set_minimized();
+}
+
+mf::XdgToplevelUnstableV6* mf::XdgToplevelUnstableV6::from(wl_resource* surface)
+{
+    auto* tmp = wl_resource_get_user_data(surface);
+    return static_cast<XdgToplevelUnstableV6*>(static_cast<wayland::XdgToplevelV6*>(tmp));
+}
+
+// XdgPositionerUnstableV6
+
+mf::XdgPositionerUnstableV6* mf::XdgPositionerUnstableV6::from(wl_resource* resource)
+{
+    auto* tmp = wl_resource_get_user_data(resource);
+    return static_cast<XdgPositionerUnstableV6*>(static_cast<wayland::XdgPositionerV6*>(tmp));
+}
+
+mf::XdgPositionerUnstableV6::XdgPositionerUnstableV6(struct wl_client* client, struct wl_resource* parent, uint32_t id)
+    : wayland::XdgPositionerV6(client, parent, id)
+{}
+
+void mf::XdgPositionerUnstableV6::destroy()
+{
+    wl_resource_destroy(resource);
+}
+
+void mf::XdgPositionerUnstableV6::set_size(int32_t width, int32_t height)
+{
+    base.size = geom::Size{width, height};
+}
+
+void mf::XdgPositionerUnstableV6::set_anchor_rect(int32_t x, int32_t y, int32_t width, int32_t height)
+{
+    base.aux_rect = geom::Rectangle{{x, y}, {width, height}};
+}
+
+void mf::XdgPositionerUnstableV6::set_anchor(uint32_t anchor)
```

Comment:
Need to pick up recent fixes.

This is the trouble with trying to replacing xdg_shell_v6.cpp with new xdg_shell_unstable_v6.cpp. Landing a bunch of "new stuff" and deleting "old stuff" where we don't (yet) have test coverage means we need to be vigilant.

Reply:
Recent changes have been merged correctly. Took a while because I realized there were flaws in how I was handling the event sink, which I addressed.